### PR TITLE
Feature/f 154 legend color hovering

### PR DESCRIPTION
**Pull request by ahmedfarouk2000**
Hover effect on legend gradient colors as defined in old HungerMap.

image

### CODE REVIEW

File: src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -23,11 +25,11 @@ export function Tooltip({ children, title, text, delay, warning }: TooltipProps)
...
   const tooltipContent = title ? (
     <div>
-      <h3 className="text-small font-bold mb-1"> {title} </h3>
-      <p className="text-small font-normal"> {text} </p>
+      <h3 className={`text-small font-bold mb-1 ${titleStyle}`}> {title} </h3>
+      <p className={`text-small font-normal ${textStyle}`}> {text} </p>
     </div>
   ) : (
-    <p className="text-small font-normal"> {text} </p>
+    <p className={`text-small font-normal ${textStyle}`}> {text} </p>
```
Comment (Tschonti) [inline src/components/Tooltip/Tooltip.tsx:32]: maybe use `clsx()` here?

File: src/components/Legend/GradientLegend.tsx [hunk 2]
```diff
@@ -20,12 +25,28 @@ export default function GradientLegend({ colors, startLabel, endLabel, hasNotAna
...
+            key={colorData.color}
+            className="absolute top-0 left-0 h-full"
+            style={{
+              width: `${segmentWidth}%`,
+              left: `${index * segmentWidth}%`,
+            }}
+          >
+            <Tooltip title={colorData.title} text={colorData.value} titleStyle="text-center" textStyle="text-center">
+              <div
+                className={`hover:bg-opacity-25 hover:bg-black group flex-1 h-full cursor-pointer ${index === 0 ? 'rounded-l-full' : ''} ${index === colorsData.length - 1 ? 'rounded-r-full' : ''}`}
```
Comment (bohdangarchu) [inline src/components/Legend/GradientLegend.tsx:40]: clsx

**Review by bohdangarchu — CHANGES_REQUESTED** [review]
pls use clsx otherwise good job!

File: src/components/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (ahmedfarouk2000) [reply src/components/Tooltip/Tooltip.tsx]: yeah sure

File: src/components/Legend/GradientLegend.tsx (hunk 2 above)
Comment (ahmedfarouk2000) [reply src/components/Legend/GradientLegend.tsx]: done

**Review by bohdangarchu — APPROVED** [review]

**Pull request merged**